### DISCUSSION TITLE
Why can't we specify *any* key for the split when dumping?

### DISCUSSION THREAD

**Discussion by texorama** [mydumper/mydumper#1653] [Ideas]
Why can we only use the primary key for splitting into threads when dumping? Is it because the physical store is determined by the primary, so we lose much of the speed benefit when dumping using a non-primary?
Otherwise, it would be great to have the option.
I'm using the weird primary keys from https://wordpress.org/plugins/index-wp-mysql-for-speed/ and I don't want to miss out on the very real operational performance benefits they bring, but they prohibit splitting. 
Thanks guys!

**Comment by davidducos** [comment]
Hi @texorama , we already have that option, as since a long time ago, mydumper is able to take the key with the higher cardinality. However, if it is not an integer, it will not be use as (for now) we are only able to split by integer keys.
